tensorboard/plugins/hparams/api.py

```diff
@@ -412,3 +413,92 @@ def as_proto(self):
         description=self._description,
         dataset_type=self._dataset_type,
     )
+
+
+class KerasCallback(tf.keras.callbacks.Callback):
+  """Callback for logging hyperparameters to TensorBoard.
+
+  NOTE: This callback only works in TensorFlow eager mode.
+  """
+
+  def __init__(
+      self,
+      logdir,
+      hparams,
+      group_name=None,
+  ):
+    """Create a callback for logging hyperparameters to TensorBoard.
+
+    Each callback object is good for one session only.
+
+    Args:
+      logdir: The log directory for this session.
```

Comment:
Yes, I think del self._writer should be sufficient if we want to preserve that the callback is one-time-use only.

If we wanted the callback to be usable across sessions, I think it would be reasonable to interpret the ownership as taking no ownership of a passed-in writer (we retain a reference for re-use, and to purge the reference you must del the callback) and taking per-session ownership of a writer constructed from a logdir string.